rclpy/rclpy/executors.py

```diff
@@ -415,7 +419,10 @@ async def handler(entity, gc, is_shutdown, work_tracker):
                 gc.trigger()
 
                 try:
-                    await call_coroutine(entity, arg)
+                    if isinstance(arg, tuple):
+                        await call_coroutine(entity, *arg)
+                    else:
+                        await call_coroutine(entity, arg)
```

Comment:
This is sort of a hack, maybe I should modify all `take_*` messages to always return a tuple of arguments, and here we could always use `await call_coroutine(entity, *arg)`

Reply:
Actually, this change broke services, actions and waitables.
I'll take a look on Monday

Reply:
Fixed in https://github.com/ros2/rclpy/pull/922/commits/73f9ace78a3f23d1fb5eaba50504d669772be126